### DISCUSSION TITLE
How can I show a column from a related table on a View form?

### DISCUSSION THREAD

**Discussion by hackel** [filamentphp/filament#5876] [Help]
I'm trying to figure out how to display a related column on a view-only form.  Currently the only way I have been able to do this is with a `Select::relationship()`, but this loads *all* the related rows as options, and I can't do that as I have a table with many thousands of rows.  Basically I would like a read-only field where I can display the related data in the same way that I can for a table column, e.g. `TextColumn::make('relationship.columnName')`.  Unfortunately this does not work for `TextInput`.  Is something like this currently possible?

**Comment by hackel** [comment]
I was able to achieve the desired result like this, but it feels a bit hacky:
```php
Forms\Components\TextInput::make('status.message')
    ->afterStateHydrated(function (Forms\Components\TextInput $component, Request $record) {
        $component->state($record->status?->message);
    }),
```

**Reply by danharrin** [reply]
https://filamentphp.com/docs/2.x/forms/layout#saving-data-to-relationships

**Reply by hackel** [reply]
Thanks @danharrin, that works, but do you know if there's any way to do it without wrapping the (in my case single) field in a Fieldset?

**Reply by danharrin** [reply]
It works with Group or Grid or Card instead
